src/compiler/checker.ts

```diff
@@ -10866,7 +10885,7 @@ namespace ts {
         }
 
         function checkStringLiteralExpression(node: StringLiteral): Type {
-            const contextualType = getContextualType(node);
+            const contextualType = getLiteralContextualType(node);

```

Comment:
Does this even need to be a type? I mean, do you actually use the type for anything, or just check if it exists? I feel like a boolean is more appropriate.